cmd/local/ttopcollect/ttopcollect.go

```diff
@@ -0,0 +1,160 @@
+//	Copyright 2023 Dremio Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ttopcollect
+
+import (
+	"bufio"
+	"embed"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/dremio/dremio-diagnostic-collector/cmd/local/conf"
+	"github.com/dremio/dremio-diagnostic-collector/pkg/simplelog"
+)
+
+type TtopService interface {
+	StartTtop(int, int) error
+	KillTtop() (string, error)
+}
+
+//go:embed lib/sjk.jar
+var f embed.FS
+
+// Ttop provides access to the ttop sjk.jar application
+type Ttop struct {
+	cmd    *exec.Cmd
+	tmpDir string
+	output []byte
+	mu     sync.Mutex // Mutex to protect concurrent access to p.output
+	tmpMu  sync.Mutex //mutext for tmpDir
+}
+
+func (t *Ttop) StartTtop(interval, pid int) error {
+	if interval == 0 {
+		return errors.New("invalid interval of 0 seconds")
+	}
+	t.tmpMu.Lock()
+	defer t.tmpMu.Unlock()
+	tmpDir, err := os.MkdirTemp("", "ddc-sjkttop")
+	if err != nil {
+		return err
+	}
+	t.tmpDir = tmpDir
+	data, err := fs.ReadFile(f, filepath.Join("lib", "sjk.jar"))
+	if err != nil {
+		return err
+	}
+
+	sjk := filepath.Join(t.tmpDir, "sjk.jar")
+	if err := os.WriteFile(sjk, data, 0600); err != nil {
+		return err
+	}
+
+	t.cmd = exec.Command("java", "-jar", sjk, "ttop", "-ri", fmt.Sprintf("%vs", interval), "-n", "100", "-p", fmt.Sprintf("%v", pid))
```

Comment:
100 threads? I personally think its a bit much but maybe you have a reason?

Reply:
it is but it approaches getting all threads on bigger servers with 64 cores. but we could lower it

Reply:
ok thats fine good point lets keep at 100